### PR TITLE
Add "list()" to JS Client Ref

### PR DESCRIPTION
**Pull request by johndmulhausen**
What the title says; gathered descriptions from the src code, let me know if I missed anything.

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘your’ (possessive pronoun) seems less likely than ‘you’ (personal pronoun).
  ```
  ...ties that can be used as parameters for your put() operation: name String. T...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...rameters for your put() operation: name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...ens new window) format. pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```
- An apostrophe may be missing.
  ```
  ...l to put(). See the Response tab in the Examples section for an illustration...
                                             ^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...t with the following properties: before String. Specifies a date, in ISO 860...
                                             ^^^^^^
  ```

**Review by jessicaschilling — CHANGES_REQUESTED** [review]
Can you please:
- swap the order of `Parameters` and `Return value` for consistency with other methods
- take both of those out of the tab box, since they're not specific to the contents of the tabs

Thank you!

image

**Comment by johndmulhausen** [comment]
> Can you please:
> 
> * swap the order of `Parameters` and `Return value` for consistency with other methods
> * take both of those out of the tab box, since they're not specific to the contents of the tabs
> 
> Thank you!

Done!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘your’ (possessive pronoun) seems less likely than ‘you’ (personal pronoun).
  ```
  ...ties that can be used as parameters for your put() operation: name String. T...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...rameters for your put() operation: name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...ens new window) format. pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...t with the following properties: before String. Specifies a date, in ISO 860...
                                             ^^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...l to put(). See the Response tab in the Examples section for an illustration.
                                             ^^^^^^^^
  ```

File: docs/reference/client-library.md [hunk 1]
```diff
@@ -305,4 +305,75 @@ Deal {
...
+_String_. Specifies a date, in ISO 8601 format. Ensures that the call to `list()` will not return any results newer than the given date.
+:::
+
+::: details maxResults
+_Number_. Specifies the maximum number of uploads to return when calling `list()`.
+:::
+
+### Return value
+
+The return value for `list()` is the same as the return value for `status()` but with one extra propery: a string field called `name` that corresponds to the value given passed to the `name` parameter in the original call to `put()`. See the **Response** tab in the **Examples** section for an illustration.
```
Comment (alanshaw) [inline docs/reference/client-library.md:379]: This is not quite correct, the return value is an Async Iterable, and values yielded from it are the same as the return value for `status()`, with an extra `name` property.

I don't know if important, but the `created` date on these objects will be the date the user uploaded the file(s) but in the status API it's the date the CID was first seen. i.e. someone may have uploaded that same file before you.

Comment (yusefnapora) [reply docs/reference/client-library.md:379]: Good catch about the return value. We should call out that it's an async iterable - the way it's worded now suggests you'll get a plain object response.

**Review by yusefnapora — CHANGES_REQUESTED** [review]
Apart from Alan's comment about the return value, this looks good to me. Feel free to merge once that's in 👍

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘put’ seems less likely than ‘out’.
  ```
  ... can be used as parameters when calling put(): name String. The name paramet...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... as parameters when calling put(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...d the same file(s). ::: pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...t with the following properties: before String. Specifies a date, in ISO 860...
                                             ^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘put’ seems less likely than ‘out’.
  ```
  ... can be used as parameters when calling put(): name String. The name paramet...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... as parameters when calling put(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...pload the same file(s). pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...t with the following properties: before String. Specifies a date, in ISO 860...
                                             ^^^^^^
  ```

**Review by yusefnapora — APPROVED** [review]
nice!

**Comment by johndmulhausen** [comment]
Thanks for the reviews, @alanshaw , @jessicaschilling , @yusefnapora and @filecorgi !

**Pull request merged**